tokio/src/runtime/metrics/runtime.rs

```diff
@@ -529,7 +529,7 @@ cfg_net! {
                 .as_inner()
                 .io_handle
                 .as_ref()
-                .and_then(|h| h.with_io_driver_metrics(f))
+                .map(|h| f(h.metrics()))
```

Comment:
We should update this fn to return the inner metrics type, originally this was a fn because of some of the option stuff but now we should be able to return a handle. There is an open PR for the timer metrics which shows the correct impl of this fn.

Reply:
I think you're talking about `Handle::with_io_driver_metrics()`? which has been replaced by `Handle::metrics()`.

As for `RuntimeMetrics::with_io_driver_metrics()`, if we delete this function, there will be a lot of duplicate code with pattern `self.handle.as_inner().io_handle.as_ref().map(|h|h.metrics().METRIC_NAME.load(Relaxed)).unwrap_or(0)`.

Reply:
I thought we don't need to do the unwrap/map anymore since the handle contains an Arc not a Weak?

Reply:
`self.handle.as_inner().io_handle` is still an Option, `Option<crate::io::driver::Handle>`, just as:

```rust
pub fn time_driver_entry_count(&self) -> u64 {
    // unwrap_or here.
    self.handle.time_handle.as_ref().map(|h| h.metrics().entry_count.load(Relaxed)).unwrap_or(0)
}
```

Reply:
@LucioFranco ping~

Reply:
Ok yes this seems fine to me.

Reply:
So what else do I need to do here?

Reply:
Nothing here, I think.